plugin/src/main/java/org/opensearch/ml/cluster/MLCommonsClusterEventListener.java

```diff
@@ -66,6 +80,29 @@ public void clusterChanged(ClusterChangedEvent event) {
             Set<String> removedNodeIds = delta.removedNodes().stream().map(DiscoveryNode::getId).collect(Collectors.toSet());
             mlModelManager.removeWorkerNodes(removedNodeIds, false);
         } else if (delta.added()) {
+            for (DiscoveryNode node : delta.addedNodes()) {
+                // 3.1 introduces a new index for the job scheduler to track jobs
+                // the statsCollectorJob needs to be run when a cluster is started with the stats settings enabled
+                // As a result, we need to wait for a data node to come up before creating the new jobs index
+                if (node.isDataNode() && Version.V_3_1_0.onOrAfter(node.getVersion())) {
+                    if (mlFeatureEnabledSetting.isMetricCollectionEnabled() && mlFeatureEnabledSetting.isStaticMetricCollectionEnabled()) {
+                        mlIndicesHandler.initMLJobsIndex(ActionListener.wrap(success -> {
+                            if (success) {
+                                mlTaskManager.startStatsCollectorJob();
+                            }
+                        }, e -> log.error("Failed to initialize ML jobs index", e)));
+                    }
+
+                    if (clusterService.state().getMetadata().hasIndex(TASK_POLLING_JOB_INDEX)) {
+                        mlIndicesHandler.initMLJobsIndex(ActionListener.wrap(success -> {
```

Comment:
Seems line 97-101 duplicate with line 89-93?

Reply:
Actually the `startTaskPollingJob()` method already have `mlIndicesHandler.initMLJobsIndex`